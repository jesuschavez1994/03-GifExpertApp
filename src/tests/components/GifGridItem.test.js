
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';
import '@testing-library/jest-dom';

describe('Pruebas en <GifGridItem />', () => {

    const title = 'Title';
    const url = 'https://localhost/img.jpg'
    let wrapper = shallow( <GifGridItem title = { title } url = { url }/>);

    test('Debe de mostrar <GifGridItem /> ', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de tener un parrafo con el title ', () => {
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
    });

    test('debe de tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
    });

    test('debe de tener la clase card ', () => {
        const div =  wrapper.find('div');
        expect( div.prop('className') ).toBe( 'card' );
    })
    
    
    
    
})
