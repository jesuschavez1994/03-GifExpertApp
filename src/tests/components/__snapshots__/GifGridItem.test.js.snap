// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe de mostrar <GifGridItem />  1`] = `
<div
  className="card"
>
  <img
    alt="Title"
    src="https://localhost/img.jpg"
  />
  <p>
    Title
  </p>
</div>
`;
